### PR TITLE
docs: add documentation for using layers with private repos

### PR DESCRIPTION
**Pull request by franklin-tina**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Update **Layers** documentation to include instructions for extending from private repositories.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.getting-started/9.layers.md [hunk 1]
```diff
@@ -29,6 +29,18 @@ export default defineNuxtConfig({
...
+
+```ts twoslash [nuxt.config.ts]
+export default defineNuxtConfig({
+  extends: [
+    // per layer configuration
+    ['github:my-themes/private-awesome', { giget: { auth: process.env.GH_TOKEN } }]
+  ]
+})
+```
+This uses [giget](https://unjs.io/packages/giget) under the hood for authenticating access to private repositories. For more information about this, visit [the documentation](https://unjs.io/packages/giget#providing-token-for-private-repositories) on "Providing token for private repositories"
```
Comment (pi0) [inline docs/1.getting-started/9.layers.md:42]:
```suggestion
This uses [unjs/giget](https://giget.unjs.io) under the hood for authenticating access to private repositories. For more information about this, visit [the documentation](https://unjs.io/packages/giget#providing-token-for-private-repositories) on "Providing token for private repositories"
```

Comment (danielroe) [inline docs/1.getting-started/9.layers.md:42]:
```suggestion

This uses [giget](https://unjs.io/packages/giget) under the hood for authenticating access to private repositories. For more information about this, visit [the documentation](https://unjs.io/packages/giget#providing-token-for-private-repositories) on "Providing token for private repositories"
```

File: docs/1.getting-started/9.layers.md [hunk 2]
```diff
@@ -29,6 +29,18 @@ export default defineNuxtConfig({
 })
 ```
 
+The above example illustrates extending from a public GitHub repository. To extend from a **private** GitHub repository, follow the modified approach outlined below.
```
Comment (danielroe) [inline docs/1.getting-started/9.layers.md:32]:
```suggestion
You can also pass an authentication token if you are extending from a private GitHub repository:
```

**Review by danielroe — APPROVED** [review]
Thank you! ❤️

File: docs/1.getting-started/9.layers.md [hunk 3]
```diff
@@ -29,6 +29,19 @@ export default defineNuxtConfig({
 })
 ```
 
+You can also pass an authentication token if you are extending from a private GitHub repository:
+
+```ts twoslash [nuxt.config.ts]
+export default defineNuxtConfig({
+  extends: [
+    // per layer configuration
+    ['github:my-themes/private-awesome', { giget: { auth: process.env.GH_TOKEN } }]
```
Comment (pi0) [inline docs/1.getting-started/9.layers.md:38]: Let's wait for https://github.com/unjs/c12/issues/142 before adding docs

File: docs/1.getting-started/9.layers.md [hunk 4]
```diff
@@ -29,6 +29,19 @@ export default defineNuxtConfig({
...
+  ]
+})
+```
+
+This uses [giget](https://unjs.io/packages/giget) under the hood for authenticating access to private repositories. For more information about this, visit [the documentation](https://unjs.io/packages/giget#providing-token-for-private-repositories) on "Providing token for private repositories"
```
Comment (pi0) [inline docs/1.getting-started/9.layers.md:43]:
```suggestion
Nuxt uses [unjs/c12](https://c12.unjs.io) and [unjs/giget](https://giget.unjs.io) for extending remote layers. Check the documentation for more information and all available options.
```

File: docs/1.getting-started/9.layers.md (hunk 3 above)
Comment (pi0) [inline docs/1.getting-started/9.layers.md:38]:
```suggestion
    ['github:my-themes/private-awesome', { auth: process.env.GITHUB_TOKEN }]
```

Comment (pi0) [reply docs/1.getting-started/9.layers.md:38]: https://github.com/unjs/c12/releases/tag/v1.10.0

**Comment by Atinux** [comment]
Thank you @franklin-tina 💚

**Pull request merged**